### PR TITLE
Add JSDoc documentation

### PR DESCRIPTION
**Pull request by timmathews**
A first pass at documentation

### CODE REVIEW

**Comment by tkurki** [comment]
I think this was there beforehand, but should be fixed:
```
tjk:signalk-js-client tjk$ npm run doc

> signalk-client@0.0.1-4 doc /Users/tjk/git-workspace/signalk/signalk-js-client
> jsdoc -r lib

ERROR: Unable to find the source file or directory /Users/tjk/git-workspace/signalk/signalk-js-client/lib
There are no input files to process.
```
How do you publish jsdoc in Github?

**Comment by timmathews** [comment]
I think we would need to have  a gh-pages branch configured to publish to signalk.org/signalk-js-client and add a doc:publish script to automate the process. Not sure about styling to make it look like gitbook.

**Comment by tkurki** [comment]
I would rather keep the docs right next to the readme and code in Github proper, not a separate web page.

How about https://github.com/jsdoc2md/jsdoc-to-markdown ?

**Comment by timmathews** [comment]
That'll work

File: index.js [hunk 1]
```diff
@@ -129,7 +150,48 @@ function Client(options) {
...
+        this.wsPath = wsUrl.href + '?subscribe=none';
+        paths.push({path: subscriptionType});
+      }
+    } else {
+      Array.prototype.slice.call(params).forEach(function(p) {
+        paths.push({path: p});
+      });
+    }
+
+    const ws = new WebSocket(this.wsPath);
```
Comment (tkurki) [inline index.js]: This will now always open the ws connection?

File: index.js [hunk 2]
```diff
@@ -129,7 +150,48 @@ function Client(options) {
...
+    ws.onopen = onOpen;
+    ws.onmessage = onMessage;
+    ws.onclose = onClose;
+    ws.onerror = onError;
+
+    ws.send(JSON.stringify({context: 'vessels.self', subscribe: paths}));
```
Comment (tkurki) [inline index.js]: The connection is not yet open i think.

**Comment by tkurki** [comment]
What's the plan for this PR? It is not just jsdoc, but also a bunch of tests and conversion to ES6. And the diff for index.js looks....challenging for a reviewer.

**Comment by timmathews** [comment]
This branch has diverged *a lot* from the initial intention. Closing the PR and renaming the branch.

**Pull request closed without merging**